### PR TITLE
[STC-128] DesktopModal

### PR DESCRIPTION
**Pull request by michellehorn**
- [x] Modal desktop
- [x] Tests
- [x] Scroll 

https://user-images.githubusercontent.com/18689578/132890981-259ed4cb-a6a6-4db8-b1cb-c4c97c17a217.mov

### CODE REVIEW

File: src/components/DesktopModal/DesktopModal.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if (isVisible) {
+      body[0].style.overflowY = "hidden"
+    } else {
+      body[0].style.overflowY = "scroll"
+    }
+  }, [isVisible])
+
+  return (
+    <Modal
+      {...className}
```
Comment (giubatt) [inline src/components/DesktopModal/DesktopModal.tsx:30]: I don't understand this spread, could you explain the reason?

File: src/components/DesktopModal/DesktopModal.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          bottom: "auto",
+          marginRight: "-50%",
+          transform: "translate(-50%, -50%)",
+          borderRadius: 0,
+          backgroundColor: "transparent",
+        },
+      }}
+      isOpen={isVisible}
+    >
+      <Paper padding={0} sx={{ zIndex: 10, width: widthModal, gap: 5 }}>
```
Comment (giubatt) [inline src/components/DesktopModal/DesktopModal.tsx:52]: I think maybe the className the DesktopModal receives could be added to the paper here (`<Paper className={className} ...etc >`), so there's no need for the widthModal prop, you could just use it like this:
```
<DesktopModal sx={{ width: 500 }}>
  <p>modal body</p>
</DesktopModal>
```

File: src/components/DesktopModal/DesktopModal.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <div sx={{ display: "grid", gap: 6, p: 8, pb: 12, minWidth: 500 }}>
+          {children}
+        </div>
```
Comment (giubatt) [inline src/components/DesktopModal/DesktopModal.tsx:55]: I think we could give more control to the component user if the children wasn't wrapped by this div, what if the use-case needs different paddings or I don't want it to be in a display grid, or the minWidth to be 500?
```suggestion
        {children}
```

**Review by giubatt — CHANGES_REQUESTED** [review]
Couple of things I couldn't find a line to add the comment so commenting here:
1. As we use yarn we shouldn't have a `package-lock.json` as that could be confusing, better to remove it from the changes
2. The [DesktopModal image snapshot](https://github.com/micromedio/herz-ui/pull/144/files#diff-42f73ba1e1fb7a3b26b18ac7bd3ff05ea70691da499d393061b820d6ffe375a9) is basically an empty image, I think you may need to add it to the `fullPageComponents` array in `Storyshots.test.ts`

**Review by giubatt — DISMISSED** [review]

**Review by VagnerNico — DISMISSED** [review]

**Review by giubatt — DISMISSED** [review]

**Review by VagnerNico — APPROVED** [review]

**Review by giubatt — DISMISSED** [review]

**Review by giubatt — DISMISSED** [review]

**Review by VagnerNico — DISMISSED** [review]

**Review by giubatt — DISMISSED** [review]

**Review by VagnerNico — APPROVED** [review]

**Pull request merged**